OutOfSchool/OutOfSchool.BusinessLogic/Services/ThumbnailProcessor/ThumbnailProcessingService.cs

```diff
@@ -0,0 +1,126 @@
+﻿using Microsoft.Extensions.Options;
+using OutOfSchool.BusinessLogic.Models;
+using OutOfSchool.ExternalFileStore;
+using OutOfSchool.ExternalFileStore.Models;
+using SkiaSharp;
+
+namespace OutOfSchool.BusinessLogic.Services.ThumbnailProcessor;
+public class ThumbnailProcessingService : IThumbnailProcessingService
+{
+    private readonly IImageService imageService;
+    private readonly IImageStorage imageStorage;
+    private readonly ILogger<ThumbnailProcessingService> logger;
+    private readonly ThumbnailGenerationOptions options;
+    public ThumbnailProcessingService(
+        IImageService service,
+        IImageStorage imageStorage,
+        ILogger<ThumbnailProcessingService> logger,
+        IOptions<ThumbnailGenerationOptions> options)
+    {
+        this.imageService = service;
+        this.imageStorage = imageStorage;
+        this.logger = logger;
+        this.options = options.Value;
+    }
+    public async Task<bool> HasThumbnail(string imageId)
+    {
+        var thumbnailId = GetThumbnailId(imageId);
+        var thumbnail = await imageService.GetByIdAsync(thumbnailId);
+
+        if (thumbnail.Succeeded)
+        {
+            return thumbnail.Value != null;
+        }
+        else
+        {
+            return false;
+        }
+    }
+
+    private string GetThumbnailId(string imageId)
+    {
+        var decoded = Uri.UnescapeDataString(imageId);
+        return decoded + $"-thumbnail.{options.Format}";
```

Comment:
do not add `.jpg` or any other 'file extension' at the end for the key name, as it would make it hard to change encoding format without changing all the keys (though it does not really matter for browsers, but i had cases when few libraries processed 'file names' and content type separately and it would be a pain to debug).

File encoding is handled by the browsers and most libraries via the `ContentType`.